NitroxClient/GameLogic/StorageSlots.cs

```diff
@@ -0,0 +1,131 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using NitroxClient.Communication.Abstract;
+using NitroxClient.GameLogic.Helper;
+using NitroxClient.GameLogic.Spawning;
+using NitroxClient.Unity.Helper;
+using NitroxModel.DataStructures.GameLogic;
+using NitroxModel.DataStructures.Util;
+using NitroxModel.Helper;
+using NitroxModel.Logger;
+using NitroxModel.Packets;
+using UnityEngine;
+
+namespace NitroxClient.GameLogic
+{
+    public class StorageSlots
+    {
+        private readonly IPacketSender packetSender;
+        private readonly LocalPlayer localPlayer;
+
+        public StorageSlots(IPacketSender packetSender, LocalPlayer localPlayer)
+        {
+            this.packetSender = packetSender;
+            this.localPlayer = localPlayer;
+        }
+
+
+        public void BroadcastItemAdd(InventoryItem item, GameObject gameObject)
+        {
+            string guid = GuidHelper.GetGuid(gameObject);
+            NitroxEntity entity = gameObject.GetComponent<NitroxEntity>();
+            if (entity == null)
+            {
+                return;
```

Comment:
I'm not sure that we can rely on NitroxEntity (btw this seems like a silent failure case).  It's original purpose was to mark items so we know what was authoritatively spawned but has kind've scope creeped to all entities.  It may be something we want to deprecate because it is losing meaning.

Reply:
Its for skipping the default batteries. We could info log that.
I need a way to distiguish synched batteries from unsynched ones (default spawned batteries issue). Maybe we have a better way doing that but i do not know how.

Reply:
Maybe an overload to the GuidHelper.GetGuid can help. I had the same problem.
The GetGuid generates a new uniqueguid if it can't find one. 
Suggestion:
.GetGuid(gameobject, bool allowEmpty) > Returns emtpy.string if no guid found. So you can see if it is a synced one.